ls1intum/Artemis
# Fix issues with plagiarism detection on programming exercises

### Checklist

### Description


Plagiarism detection currently has some problems for multiple reasons. This PR should resolve the following issues in sentry:
- https://sentry.ase.in.tum.de/organizations/artemis/issues/634 `MysqlDataTruncation: Data truncation: Data too long for column 'file' at row 1`
  - Fix: extend the maximum length for the file column to 255 characters. I can't imagine anyone pushing files with longer names without malicious intentions. @krusche I suggest creating a pre-receive hook in bitbucket that rejects longer filenames. If you want me to do that ping me.
- https://sentry.ase.in.tum.de/organizations/artemis/issues/796 `Malformed input or input contains unmappable characters: repos-download /path/to/repo/with/奇怪的/file.name`
  - Fix: I additionally catch a InvalidPathException to prevent the plagiarism detection request to fail. In a future PR I'll implement a notification to the Instructor about which repos could not be checked for plagiarism.
- https://sentry.ase.in.tum.de/organizations/artemis/issues/479
  - when there is no plagiarism result yet the client tries to access a index on a list that is not present. I implemented additional checks.

### Steps for Testing


1. Submit a programming excercise as two different students. Your sumbission should contain a file with a path name longer than 50 characters. I prepared this step on ts1 -> Joscha Henningsens Test Course -> Plagiatstest
2. As instructor (re)check this exercise for plagiarism. Observe the network calls in the developer console. The request `check-plagiarism` must **not** fail.
